test/functional/feature_config_args.py

```diff
@@ -78,7 +78,7 @@ def test_config_file_parser(self):
             conf.write('')  # clear
 
     def test_log_buffer(self):
-        with self.nodes[0].assert_debug_log(expected_msgs=['Warning: parsed potentially confusing double-negative -connect=0']):
+        with self.nodes[0].assert_debug_log(expected_msgs=['Warning: parsed potentially confusing double-negative -noconnect=0']):
```

Comment:
In commit "Revamp option negating policy" (3f0b5e28ebcf42c168f393a3109403354cb8e128)

This commit appears to be changing behavior by fixing the error message so it technically contradicts "This PR is only a refactoring and does not change behavior" in the PR description. Could consider removing this here and just doing it as a minor bugfix later. Or updating the PR description to mention the change in behavior.

Reply:
> Could consider removing this here and just doing it as a minor bugfix later.

Done.